client/analytics/settings/historical-data/index.js

```diff
@@ -0,0 +1,174 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import HistoricalDataActions from './actions';
+import HistoricalDataPeriodSelector from './period-selector';
+import HistoricalDataProgress from './progress';
+import HistoricalDataStatus from './status';
+import HistoricalDataSkipCheckbox from './skip-checkbox';
+import withSelect from 'wc-api/with-select';
+import './style.scss';
+
+class HistoricalData extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.dateFormat = __( 'MM/DD/YYYY', 'woocommerce-admin' );
+
+		this.state = {
+			period: {
+				date: moment().format( this.dateFormat ),
+				label: 'all',
+			},
+			skipChecked: true,
+		};
+
+		this.onDateChange = this.onDateChange.bind( this );
+		this.onPeriodChange = this.onPeriodChange.bind( this );
+		this.onSkipChange = this.onSkipChange.bind( this );
+	}
+
+	onPeriodChange( val ) {
+		this.setState( {
+			period: {
+				...this.state.period,
+				label: val,
+			},
+		} );
+	}
+
+	onDateChange( val ) {
+		this.setState( {
+			period: {
+				date: val,
+				label: 'custom',
+			},
+		} );
+	}
+
+	onSkipChange( val ) {
+		this.setState( {
+			skipChecked: val,
+		} );
+	}
+
+	getStatus() {
+		const {
+			customersProgress,
+			customersTotal,
+			inProgress,
+			ordersProgress,
+			ordersTotal,
+		} = this.props;
+
+		if ( inProgress ) {
+			if ( customersProgress < customersTotal ) {
+				return 'customers';
```

Comment:
Side note here- orders can create customers on the fly without first importing customers.  https://github.com/woocommerce/woocommerce-admin/pull/2034 may remove the initial customer import (at least when days are specified).

It probably doesn't affect this and I don't think any changes are needed now, but just wanted to point this out for reference in case we run into any issues later on.

Reply:
Thanks for pointing this out @joshuatf! I'm pretty sure we will have to do some more tweaks once we start connecting this with the real endpoints.